### PR TITLE
added github leaderboard endpoint

### PR DESCRIPTION
**Pull request by elimelt**
this one is reaaaal slow, we should really figure out best way for cron

### CODE REVIEW

File: server/leaderboard/management/commands/update_github_stats.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    def get_github_stats(self, username):
+        """Fetch GitHub statistics using public API and web scraping for contributions"""
+        try:
+            # contribution count
+            contributions_url = f"https://github.com/users/{username}/contributions"
+            response = requests.get(contributions_url)
+            contributions = 0
+
+            if response.status_code == 200:
+                soup = BeautifulSoup(response.text, 'html.parser')
```
Comment (ShawnCollinge) [inline server/leaderboard/management/commands/update_github_stats.py:39]: gross.

File: server/leaderboard/management/commands/update_github_stats.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                timeout=10
+            )
+
+            if user_response.status_code != 200:
+                logger.error(f"Failed to fetch user data for {username}: {user_response.status_code}")
+                return None
+
+            user_data = user_response.json()
+
+            # PR count
```
Comment (ShawnCollinge) [inline server/leaderboard/management/commands/update_github_stats.py:58]: do we need this comment?

File: server/leaderboard/management/commands/update_github_stats.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            total_prs = 0
+            if pr_response.status_code == 200:
+                pr_data = pr_response.json()
+                total_prs = pr_data.get("total_count", 0)
+            else:
+                logger.warning(f"Failed to fetch PR data for {username}: {pr_response.status_code}")
+
+            return {
+                "total_prs": total_prs,
+                "total_commits": contributions,  # contributions count from profile
```
Comment (ShawnCollinge) [inline server/leaderboard/management/commands/update_github_stats.py:74]: I mean that's obvious right?

File: server/leaderboard/views.py [hunk 4]
```diff
@@ -65,3 +65,39 @@ def get_queryset(self):
...
+        # time filter
+        if time_range:
+            try:
+                hours = int(time_range)
+                cutoff = timezone.now() - timedelta(hours=hours)
+                queryset = queryset.filter(last_updated__gte=cutoff)
+            except ValueError:
+                raise ValidationError("updated_within must be a valid number of hours")
+
+        # ordering
```
Comment (ShawnCollinge) [inline server/leaderboard/views.py:89]: lol really? I didn't know ordering_options would be ordering.

**Review by ShawnCollinge — APPROVED** [review]

File: server/leaderboard/management/commands/update_github_stats.py (hunk 1 above)
Comment (elimelt) [reply server/leaderboard/management/commands/update_github_stats.py:39]: it is the only way

File: server/leaderboard/management/commands/update_github_stats.py (hunk 2 above)
Comment (elimelt) [reply server/leaderboard/management/commands/update_github_stats.py:58]: bro

File: server/leaderboard/management/commands/update_github_stats.py (hunk 3 above)
Comment (elimelt) [reply server/leaderboard/management/commands/update_github_stats.py:74]: hates

File: server/leaderboard/views.py (hunk 4 above)
Comment (elimelt) [reply server/leaderboard/views.py:89]: comments

Comment (elimelt) [reply server/leaderboard/views.py:89]: i like to be able to glance quickly at where logical sections start and stop, TBS these aren't super long yet so it isn't doing much

Comment (elimelt) [reply server/leaderboard/views.py:89]: you're just trying to pad your comment stat in case I add it to leaderboard 😂

**Pull request merged**